easybuild/tools/module_naming_scheme/hierarchical_mns.py

```diff
@@ -164,18 +166,28 @@ def det_modpath_extensions(self, ec):
             comp_name_ver = None
             if ec['name'] in extend_comps:
                 for key in COMP_NAME_VERSION_TEMPLATES:
-                    if ec['name'] in key.split(','):
-                        comp_name, comp_ver_tmpl = COMP_NAME_VERSION_TEMPLATES[key]
-                        comp_versions = {ec['name']: self.det_full_version(ec)}
-                        if ec['name'] == 'ifort':
-                            # 'icc' key should be provided since it's the only one used in the template
-                            comp_versions.update({'icc': self.det_full_version(ec)})
-                        if tc_comp_info is not None:
-                            # also provide toolchain version for non-dummy toolchains
-                            comp_versions.update({tc_comp_info[0]: tc_comp_info[1]})
-
-                        comp_name_ver = [comp_name, comp_ver_tmpl % comp_versions]
-                        break
+		    if tc_comp_info is not None:
+			if ec['name'] in key.split(',') and tc_comp_info[0] in key.split(','):
+			    comp_name, comp_ver_tmpl = COMP_NAME_VERSION_TEMPLATES[key]
+			    comp_versions = {ec['name']: self.det_full_version(ec)}
+			    if ec['name'] == 'ifort':
+				# 'icc' key should be provided since it's the only one used in the template
+				comp_versions.update({'icc': self.det_full_version(ec)})
+			    # also provide toolchain version for non-dummy toolchains
+			    comp_versions.update({tc_comp_info[0]: tc_comp_info[1]})
+
+			    comp_name_ver = [comp_name, comp_ver_tmpl % comp_versions]
+			    break
+		    else:
+			if ec['name'] in key.split(','):
+			    comp_name, comp_ver_tmpl = COMP_NAME_VERSION_TEMPLATES[key]
+			    comp_versions = {ec['name']: self.det_full_version(ec)}
+			    if ec['name'] == 'ifort':
+				# 'icc' key should be provided since it's the only one used in the template
+				comp_versions.update({'icc': self.det_full_version(ec)})
```

Comment:
@akesandgren there's a lot of copy-pasting going on here, not a big fan of that

Also, I'd like to see this handled through a separate PR, preferably combined with #1753 which we should push through as well.
There's still an open question holding back that PR, but I feel that shouldn't be held back much longer...